#! /bin/sh -e
## 03_qt_fontlist_fix.dpatch by Martin Loschwitz <madkiss@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix fonts list to show only the fonts we can use

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0

diff -ruN qt-x11-free-3.2.1.old/src/kernel/qfontdatabase_x11.cpp qt-x11-free-3.2.1/src/kernel/qfontdatabase_x11.cpp
--- qt-x11-free-3.2.1.old/src/kernel/qfontdatabase_x11.cpp	2003-08-20 15:07:34.000000000 +0200
+++ qt-x11-free-3.2.1/src/kernel/qfontdatabase_x11.cpp	2003-09-03 11:08:05.000000000 +0200
@@ -673,6 +673,9 @@
     if ( fontFamily && fontFamily->xlfdLoaded )
 	return;
 
+#ifdef QT_XFT2
+    if ( !qt_has_xft ) {
+#endif // QT_XFT2
     int fontCount;
     // force the X server to give us XLFDs
     QCString xlfd_pattern = "-*-";
@@ -764,8 +767,11 @@
     }
 
     XFreeFontNames( fontList );
-}
 
+#ifdef QT_XFT2
+    }
+#endif // QT_XFT2
+}
 
 #ifndef QT_NO_XFTFREETYPE
 static int getXftWeight(int xftweight)
